### PR TITLE
Eng 7491 request gps location

### PR DESCRIPTION
**Pull request by kishisaka-nid**
Add a location request listener to that will poll for locations every 10 seconds when the customer moves 10 meters. The coroutine that controls the location collection is tied to the session lifecycle. 

Will setup the coroutine when start, startSession or resumeCollection is called. 

Will start listening for locations when metadata is collected in createMobileMetadata(). 

Will stop listening and cancel the coroutine when the stop, stopSession or pauseCollection is called. 

All the integration tests pass locally and on the build box. 
image

### CODE REVIEW

**Review by priya-xavier — DISMISSED** [review]
LGTM! I'm able to see the coordinates on API 30,31 &28 simulators.

File: NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt [hunk 1]
```diff
@@ -442,7 +448,11 @@ class NeuroID
             }
             dataStore.saveAndClearAllQueuedEvents()
 
-            this.getApplicationContext()?.let { nidCallActivityListener?.setCallActivityListener(it) }
+            this.getApplicationContext()
+                ?.let { nidCallActivityListener?.setCallActivityListener(it) }
+            locationService?.let { ls ->
+                ls.setupLocationCoroutine(getApplicationContext()?.getSystemService(Context.LOCATION_SERVICE) as LocationManager)
+            }
```
Comment (kevinsites) [inline NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt:455]: is there a reason you wouldn't have the locationService.let inside the getApplicationContext you already called?

File: app/src/androidTest/java/com/sample/neuroid/us/MockServerTest.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                }
+            }
+
+            if (booleanIsFound) {
+                assert(true) {
+                    "Found the event man!"
+                }
+            } else {
+                assert(false) {
+                    "Failed to find event man!"
```
Comment (kevinsites) [inline app/src/androidTest/java/com/sample/neuroid/us/MockServerTest.kt:143]: Should we put the event type that was not found?

File: NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt (hunk 1 above)
Comment (kishisaka-nid) [reply NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt:455]: can do

File: app/src/androidTest/java/com/sample/neuroid/us/MockServerTest.kt (hunk 2 above)
Comment (kishisaka-nid) [reply app/src/androidTest/java/com/sample/neuroid/us/MockServerTest.kt:143]: can do

**Review by kevinsites — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Pull request merged**